numpy/random/_examples/cython/extending_distributions.pyx

```diff
@@ -72,3 +74,46 @@ def uint10_uniforms(Py_ssize_t n):
     randoms = np.asarray(random_values)
     return randoms
 
+# cython example 3
+def uniforms_ex(Py_ssize_t n, dtype=np.float64):
+    """
+    Create an array of `n` uniformly distributed doubles via a "fill" function.
+
+    A 'real' distribution would want to process the values into
+    some non-uniform distribution
+
+    Parameters
+    ----------
+    n: int
+        Output vector length
+    dtype: {str, dtype}, optional
+        Desired dtype, either 'd' (or 'float64') or 'f' (or 'float32'). The
+        default dtype value is 'd'
+    """
+    cdef Py_ssize_t i
+    cdef bitgen_t *rng
+    cdef const char *capsule_name = "BitGenerator"
+    cdef np.ndarray randoms
+
+    typedict = {'f': np.float32, 'd': np.float64, 'float64': np.float64,
+                'float32': np.float32, np.float32: np.float32,
+                np.float64: np.float64}
+    typ = typedict.get(dtype, None)
+    if not typ:
+        raise TypeError('Unsupported dtype "%r"' % dtype)
+    x = PCG64()
```

Comment:
Would it be a better example if this were passed in? As written, this example seems to fall into the "construct a new RNG each time" trap.

Reply:
I would like to see an example with the BitGenerator or Generator passed in, I think this would typically be best practice.

Reply:
ok